lib/rails_cursor_pagination/paginator.rb

```diff
@@ -109,21 +112,46 @@ def ensure_valid_params!(relation, first, after, last, before, order)
         raise ParameterError,
               "`order` must be either :asc or :desc, but was `#{order}`"
       end
+      if first.present? && first.negative?
+        raise ParameterError, "`first` cannot be negative, but was `#{first}`"
+      end
+      if last.present? && last.negative?
+        raise ParameterError, "`last` cannot be negative, but was `#{last}`"
+      end
+      if limit.present? && limit.negative?
+        raise ParameterError, "`limit` cannot be negative, but was `#{limit}`"
+      end
+
+      true
+    end
+
+    # @param limit [Integer, nil]
```

Comment:
I'd suggest also adding a bit of a description of what this method is doing:
```suggestion

    # Ensure that the parameters of this service are combined in a valid way.
    # Otherwise raise a +RailsCursorPagination::ParameterError+.
    #
    # @param limit [Integer, nil]
```